lib/api/include/irods/apiNumberData.h

```diff
@@ -37,8 +37,6 @@ API_NUMBER(DATA_PUT_AN,                             607)
 API_NUMBER(DATA_OBJ_GET_AN,                         608)
 API_NUMBER(DATA_GET_AN,                             609)
 API_NUMBER(DATA_COPY_AN,                            611)
-// DEPRECATED: SimpleQuery is deprecated. Use GenQuery or SpecificQuery instead.
-API_NUMBER(SIMPLE_QUERY_AN,                         614)
```

Comment:
just checking...  we remove API numbers?     leaving them here would prevent accidental reuse?

Reply:
Well, we apparently have done so in the past. Notice that API number 610 is missing. The prevention of accidental reuse is the only real reason to leave it there.

If we leave it, we cannot change the signature or the structure of the inputs and outputs (meaning we have to continue maintaining all the deprecated data structures associated with SimpleQuery). If we remove it, the caller will just an error back saying no API exists by that number. I think I prefer the latter.

Reply:
As a compromise, we could leave it here commented out (or just a descriptive comment explaining what 614 used to be).

Reply:
We also have `is_api_number_supported`.
https://github.com/irods/irods/blob/8013bd306e7b4fb54a7af6ff8695faf1e698f8f5/server/core/src/rsApiHandler.cpp#L67-L79

The XMSG API numbers are still present in apiNumberData.h.
https://github.com/irods/irods/blob/8013bd306e7b4fb54a7af6ff8695faf1e698f8f5/lib/api/include/irods/apiNumberData.h#L76-L78
https://github.com/irods/irods/blob/8013bd306e7b4fb54a7af6ff8695faf1e698f8f5/server/core/src/irods_api_number_validator.cpp#L13-L24

So we have a tool for disallowing use of the API while allowing removal of the data types and functions.

With that said, I also prefer the comment out approach. The code I linked feels like bloat and should be removed since this effort is effectively establishing the rules for how to remove an API.

Reply:
Okay, I left it in, but commented out, and included a comment saying that the API has been removed.